mrmustard/lab_dev/states/base.py

```diff
@@ -86,6 +394,102 @@ def __init__(self, name: Optional[str] = None, modes: Optional[Sequence[int]] =
         name = name or ""
         super().__init__(name, modes_out_ket=modes)
 
+    @classmethod
+    def from_bargmann(
+        cls,
+        modes: Sequence[int],
+        triple: tuple[ComplexMatrix, ComplexVector, complex],
+        name: Optional[str] = None,
+        batched: bool = False,
+    ) -> Ket:
+        A = math.astensor(triple[0])
+        b = math.astensor(triple[1])
+        c = math.astensor(triple[2])
+
+        n_modes = len(modes)
+        A_sh = (1, n_modes, n_modes) if batched else (n_modes, n_modes)
+        b_sh = (1, n_modes) if batched else (n_modes,)
+        if A.shape != A_sh or b.shape != b_sh:
+            msg = f"Given triple is inconsistent with modes=``{modes}``."
+            raise ValueError(msg)
+
+        ret = Ket(name, modes)
+        ret._representation = Bargmann(A, b, c)
+        return ret
+
+    @classmethod
+    def from_fock(
+        cls,
+        modes: Sequence[int],
+        array: ComplexTensor,
+        name: Optional[str] = None,
+        batched: bool = False,
+    ) -> Ket:
+        array = math.astensor(array)
+
+        n_modes = len(modes)
+        if len(array.shape) != n_modes + (1 if batched else 0):
+            msg = f"Given array is inconsistent with modes=``{modes}``."
+            raise ValueError(msg)
+
+        ret = Ket(name, modes)
+        ret._representation = Fock(array, batched)
+        return ret
+
+    @classmethod
+    def from_phasespace(
+        cls,
+        modes: Sequence[int],
+        cov: ComplexMatrix,
+        means: ComplexMatrix,
+        name: Optional[str] = None,
+        atol_purity: Optional[float] = 1e-3,
+    ):
+        cov = math.astensor(cov)
+        means = math.astensor(means)
+
+        n_modes = len(modes)
+        if means.shape != (2 * n_modes,):
+            msg = f"Given ``means`` is inconsistent with modes=``{modes}``."
+            raise ValueError(msg)
+        if cov.shape != (2 * n_modes, 2 * n_modes):
+            msg = f"Given ``cov`` is inconsistent with modes=``{modes}``."
+            raise ValueError(msg)
+
+        if atol_purity:
+            p = purity(cov)
+            if p < 1.0 - atol_purity:
+                msg = f"Cannot initialize a ket: purity is {p:.3f} (must be 1.0)."
+                raise ValueError(msg)
+
+        ret = Ket(name, modes)
+        ret._representation = Bargmann(*wigner_to_bargmann_psi(cov, means))
+        return ret
+
+    @property
+    def probability(self) -> float:
+        dm = self @ self.dual
+
+        rep = dm.representation
+        msg = "Method ``probability`` not supported for batched representations."
```

Comment:
@ziofil @sylviemonet 
I added these errors (and similar ones to L2) because I believe that there may be an issue with ``trace`` for ``Fock`` objects, when these objects contain batched arrays.

Below I left a minimal working example of the suspected bug. I believe that these should be addressed in a separate PR

```
from mrmustard.lab_dev.states import *
from mrmustard import settings
settings.AUTOCUTOFF_MAX_CUTOFF = 10

state = Coherent([0], 1) + Coherent([0], -1)
state.representation.trace([0], [0])  # this is fine

state = Coherent([0], 1) + Coherent([0], -1)
state = state.to_fock_component()
state.representation.trace([0], [0])  # this errors
```